lib/puppet/provider/package/brew.rb

```diff
@@ -8,13 +8,16 @@
   def install
     name = install_name
 
-    Puppet.debug "Installing #{name}"
-    output = execute([command(:brew), :install, name, *install_options])
-
-    if output =~ /Searching taps/
-      raise Puppet::ExecutionFailure, "Could not find package #{name}"
+    begin
+      Puppet.debug "Looking for #{name} package..."
+      output = execute([command(:brew), :info, name], failonfail: true)
+    rescue Puppet::ExecutionFailure => detail
+      raise Puppet::Error, "Could not find package: #{name}"
     end
 
+    Puppet.debug "Package found, installing..."
+    output = execute([command(:brew), :install, name, *install_options])
```

Comment:
Would setting this to `failonfail` be a good idea? We want some way to warn the user if a package installation failed: I'm thinking we wrap this in a `begin-rescue` like the `info` command above and raise a package installation error on fail.

Please also check to make sure the checksum check below counts as success, or move it into the `resuce` block as a potential recovery.